template/template.go

```diff
@@ -174,11 +176,11 @@ func (e *executer) resolveDeclarations(declarations []ast.Declaration) error {
 			m[d.Operand] = struct{}{}
 
 			resolved, err := e.resolveExpression(ast.Expression(d))
+			e.variables[string(d.Operand.(ast.Variable))] = resolved //nolint: forcetypeassert // Will always be a variable.
```

Comment:
Please leave a comment. This flow is unusual and deviates from the idiomatic Golang.